### PR TITLE
[TS] Fixed Input.Text style and horizontalAlignment roundtrip; added test

### PR DESCRIPTION
**Pull request by ignacionr**
As per https://github.com/Microsoft/AdaptiveCards/issues/1633 there were a couple of issues related to parse/toJSON roundtrips.

### CODE REVIEW

**Comment by ignacionr** [comment]
@dclaux if you can spare a few secs for this one; I don't think there's anybody else with some minutes to review TypeScript? For reasons too long to explain, this is blocking me for finishing the media rendering. Merci d'avance!

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 1]
```diff
@@ -347,7 +347,9 @@ export abstract class CardElement {
 
         Utils.setProperty(result, "type", this.getJsonTypeName());
         Utils.setProperty(result, "id", this.id);
-        Utils.setEnumProperty(Enums.HorizontalAlignment, result, "horizontalAlignment", this.horizontalAlignment);
+        if (this.horizontalAlignment !== null) {
```
Comment (dclaux) [inline source/nodejs/adaptivecards/src/card-elements.ts]: Blank line before if

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 2]
```diff
@@ -347,7 +347,9 @@ export abstract class CardElement {
...
+            Utils.setEnumProperty(Enums.HorizontalAlignment, result, "horizontalAlignment", this.horizontalAlignment);
+        }
```
Comment (dclaux) [inline source/nodejs/adaptivecards/src/card-elements.ts]: Blank line after closing }

File: source/nodejs/adaptivecards/src/card-elements.ts (hunk 1 above)
Comment (dclaux) [inline source/nodejs/adaptivecards/src/card-elements.ts]: I think this should be `if (this.horizontalAlignment)`

Comment (ignacionr) [reply source/nodejs/adaptivecards/src/card-elements.ts]: I thought that too. But "Left" is equivalent to 0, which is falsey.

Comment (ignacionr) [reply source/nodejs/adaptivecards/src/card-elements.ts]: Want me to shift the values and make "Left" be 1? I normally prefer to have an "Undefined"/"Invalid"/"Uninitialized" zero-equivalent on enums.

Comment (dclaux) [reply source/nodejs/adaptivecards/src/card-elements.ts]: Indeed

**Pull request merged**